Modules/RatMama.py

```diff
@@ -0,0 +1,195 @@
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! Case #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
+        if system_name.casefold() != exist_rescue.system.casefold():
+            diff_response += f"System changed! "
+
+        if platform_name.casefold() != exist_rescue.platform.name.casefold():
+            diff_response += "Platform changed! "
+
+        if not o2_status != exist_rescue.code_red:
+            diff_response += "O2 Status changed!" if o2_status else\
+                "O2 Status changed, it is now CODE RED!"
+
+        if not diff_response == "":
+            await ctx.reply(diff_response)
+
+    else:
+        platform = None
+
+        if platform_name.casefold() in ("pc", "ps", "xb"):
+            platform = Platforms[platform_name.upper()]
+        else:
+            LOG.warning(f"Got unknown platform from RatMama: {platform_name}")
+
+        # no case for that name, we have to make our own
+        rescue: Rescue = Rescue(client=client_name, system=system_name, irc_nickname=nickname,
+                                code_red=not o2_status, lang_id=lang_code, platform=platform)
+
+        board.append(rescue, overwrite=False)
+        index = board.find_by_name(client=client_name).board_index
+        await ctx.reply(f"RATSIGNAL - CMDR {client_name} - "
+                        f"Reported System: {system_name} (distance to be implemented) - "
+                        f"Platform: {platform_name} - "
+                        f"O2: {'OK' if o2_status else 'NOT OK'} - "
+                        f"Language: {result.group('full_language')}"
+                        f" (Case #{index}) ({platform_name.upper()}_SIGNAL)"
+                        )
+
+
+@rule("ratsignal", case_sensitive=False, full_message=False, pass_match=False, prefixless=True)
+async def handle_selfissued_ratsignal(ctx: Context) -> None:
+    """
+    Tries to extract as much details as possible from a self-issued ratsignal and appends
+      these details to the rescue board.
+    Should it be unable to extract the details, it will open a case and ask for the details
+      to be set and will only set the name and nick fields of the rescue
+
+    :param ctx: Context of the self-issued ratsignal
+    :return: None
+    """
+    message: str = ctx.words_eol[0]
+    # the ratsignal is nothing we are interested anymore
+    message = re.sub("ratsignal", "", message, flags=re.I)
+
+    for rescue in board.rescues.values():
+        if rescue.irc_nickname.casefold() == ctx.user.nickname.casefold():
+            await ctx.reply(
+                "You already sent a signal, please be patient while a dispatch is underway.")
+            return
+
+    sep: chr = None
```

Comment:
`chr` is not a class representing a single character, Python doesn't have one. [`chr` is a function.](https://docs.python.org/3.7/library/functions.html#chr)